core-war/src/main/java/org/silverpeas/cmis/walkers/TreeWalkerForSimpleDocument.java

```diff
@@ -158,35 +299,23 @@ protected ObjectInFolderList browseObjectsInFolder(final LocalizedResource objec
   @Override
   protected List<ObjectParentData> browseParentsOfObject(final LocalizedResource object,
       final Filtering filtering) {
-    org.silverpeas.core.contribution.attachment.model.Document document =
-        (org.silverpeas.core.contribution.attachment.model.Document) object;
+    Document document =
+        (Document) object;
     String language = filtering.getLanguage();
     ContributionIdentifier parentId = document.getSourceContribution();
-    LocalizedResource parent = findContribution(parentId.getLocalId(),
-        parentId.getComponentInstanceId());
+    LocalizedResource parent =
+        findParentContribution(parentId.getLocalId(), parentId.getComponentInstanceId());
     CmisFolder cmisParent = getCmisObject(parent, language);
-    CmisObject cmisObject = createCmisObject(document, language);
+    CmisObject cmisObject = encodeToCmisObject(document, language);
     ObjectParentData parentData = buildObjectParentData(cmisParent, cmisObject, filtering);
     return Collections.singletonList(parentData);
   }
 
-  private LocalizedResource findContribution(final String id, final String instanceId) {
-    LocalizedResource contribution;
-    try {
-      contribution = publicationService.getDetail(new PublicationPK(id, instanceId));
-    } catch (Exception e) {
-      contribution = null;
-    }
-    if (contribution == null) {
-      try {
-        contribution = nodeService.getDetail(new NodePK(id, instanceId));
-      } catch (Exception e) {
-        throw new CmisObjectNotFoundException(
-            String.format("No such parent contribution %s found in application %s", id,
-                instanceId));
-      }
-    }
-    return contribution;
+  private LocalizedResource findParentContribution(final String id, final String instanceId) {
+    String parentId = ContributionIdentifier.from(instanceId, id, CoreContributionType.UNKNOWN)
```

Comment:
A publication or a node could have exactly same local identifier into a same component instance.
It is possible here that the method returns the wrong localized parent resource.